x-pack/plugins/ml/common/types/ml_url_generator.ts

```diff
@@ -0,0 +1,187 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { RefreshInterval, TimeRange } from '../../../../../src/plugins/data/common/query';
+import { JobId } from '../../../reporting/common/types';
+import { ML_PAGES } from '../constants/ml_url_generator';
+
+export interface MLPageState<PageType, PageState> {
+  page: PageType;
+  pageState?: PageState;
+}
+
+export const ANALYSIS_CONFIG_TYPE = {
+  OUTLIER_DETECTION: 'outlier_detection',
+  REGRESSION: 'regression',
+  CLASSIFICATION: 'classification',
+} as const;
+
+type DATAFRAME_ANALYTICS_TYPE = typeof ANALYSIS_CONFIG_TYPE[keyof typeof ANALYSIS_CONFIG_TYPE];
```

Comment:
I think we only use this notation for constants. For types, it's a camel case starting with capital.
```suggestion
type DataFrameAnalyticsType = typeof ANALYSIS_CONFIG_TYPE[keyof typeof ANALYSIS_CONFIG_TYPE];
```

Reply:
Updated here https://github.com/elastic/kibana/pull/75696/commits/8248d7f0dc8ecbe9dcafa9e2eb746283723fd2df